app/src/main/java/com/lightricks/feedexercise/data/FeedRepository.kt

```diff
@@ -1,9 +1,95 @@
 package com.lightricks.feedexercise.data
 
+import android.content.Context
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.LiveDataReactiveStreams
+import androidx.lifecycle.Transformations
+import androidx.room.Room
+import com.bumptech.glide.load.Transformation
+import com.lightricks.feedexercise.database.FeedDatabase
+import com.lightricks.feedexercise.database.FeedItemDBEntity
+import com.lightricks.feedexercise.network.FeedApiService
+import com.lightricks.feedexercise.network.TemplatesMetadata
+import io.reactivex.BackpressureStrategy
+import io.reactivex.Completable
+import io.reactivex.Observable
+import io.reactivex.schedulers.Schedulers
+
 /**
  * This is our data layer abstraction. Users of this class don't need to know
  * where the data actually comes from (network, database or somewhere else).
  */
-class FeedRepository {
-    //todo: implement
+
+class FeedRepository(private var db: FeedDatabase,
+                     private var feedApiService: FeedApiService) {
+    private val urlPrefix = FeedApiService.BASE_URL + "catalog/thumbnails/"
+
+    fun setFeedApiService(feedApiService: FeedApiService) {
+        this.feedApiService=feedApiService
+    }
+
+    fun setDB(db: FeedDatabase) {
+        this.db=db
+    }
+
+    fun refresh(): Completable {
+        return feedApiService.getFeed()
+            .subscribeOn(Schedulers.io())
+            .flatMapCompletable {
+                updateFromTemplatesMetadata(it)
+            }
+    }
+
+    fun fetchData(): LiveData<List<FeedItem>>{
+        return Transformations.map(db.feedItemDao().getAll()){ it.toFeedItems() }
+        /*
+        // to use with Observable<> DAO
+        return LiveDataReactiveStreams
+            .fromPublisher(
+                db.feedItemDao().getAll()
+                    .map { it.toFeedItems() }
+                    .toFlowable(BackpressureStrategy.LATEST)
+            )
+
+         */
+    }
+
+    private fun updateFromTemplatesMetadata(list: TemplatesMetadata): Completable {
+        val resp = list?.templatesMetadata?.map { it ->
+            FeedItemDBEntity(
```

Comment:
This could also be an extension function:
`private fun List<TemplateMetadataItem>.toEntities(): List<FeedItemDBEntity> = TODO()`